### PR TITLE
Add glossary to the Logstash doc

### PR DESCRIPTION
**Pull request by dedemorton**
This PR completes the changes listed in issue #3414.

### CODE REVIEW

**Comment by untergeek** [comment]
I like it! LGTM™

**Comment by suyograo** [comment]
@dedemorton Can we add 
`@metadata`: https://www.elastic.co/guide/en/logstash/current/event-dependent-configuration.html#metadata
`conditionals`: https://www.elastic.co/guide/en/logstash/current/event-dependent-configuration.html#conditionals
`field reference`: https://www.elastic.co/guide/en/logstash/current/event-dependent-configuration.html#logstash-config-field-references

**Comment by dedemorton** [comment]
@suyograo I've added the terms plus a few more. I had a "face palm" moment this afternoon when I realized that there was another [glossary file in the repo](https://github.com/elastic/logstash/blob/master/docs/static/logstash-glossary.asciidoc) (wasn't hooked up to the TOC). IMHO, the old glossary defines a lot of terminology that isn't Logstash-specific, so I didn't include those terms, but I did try to merge (and massage) the definitions that I thought we would want to preserve. I didn't include industry terminology or descriptions of other products because that's why we have google, right? I think a shorter glossary that focuses on Logstash terminology is preferable.

BTW, I think that "worker" is better than "pipeline worker" but I can change it if you feel that pipeline worker is necessary.

File: docs/static/glossary.asciidoc [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+[[glossary-filter-plugin]]filter plugin::
+  A Logstash <<glossary-plugin,plugin>> that performs intermediary processing on an event. Typically, filters act upon
+  event data after it has been ingested via inputs, by mutating, enriching, and/or modifying the data according to
+  configuration rules. Filters are often applied conditionally depending on the characteristics of the event. Popular
+  filter plugins include grok, mutate, drop, clone, and geoip. Filter stages are optional.
+  
+[[glossary-gem]]gem::
+  A self-contained package of code that's hosted on https://rubygems.org[RubyGems.org]. You can use the Logstash plugin
+  manager to manage Lostash gems.
```
Comment (suyograo) [inline docs/static/glossary.asciidoc]: typo `Lostash`

Also can you add that Logstash plugins are packaged as Ruby Gems? Thats the connection here.

**Comment by suyograo** [comment]
@dedemorton face palm indeed. I completely forgot we had an old one lying around as well. 

Another common term used in Logstash context: 
"Message Buffer/Queue": A message buffer is an external software (Redis, Kafka, RabbitMQ) that stores messages from the Logstash shipper instance as an intermediate store, waiting to be processed by the Logstash indexer instance. This typically helps to manage bursts of data.

**Comment by dedemorton** [comment]
@suyograo We also use the term "broker" in the docs to mean (I think) the same thing. I'm a bit concerned about defining a message buffer (or queue) as external software because it's a general term that we might also want to use to describe internal queuing (for example, when we talk about things like persistent queues). How about this instead:

message broker:  Also referred to as a _message buffer_ or _message queue_, a message broker is external software (such as Redis, Kafka, or RabbitMQ) that stores messages from the Logstash shipper instance as an intermediate store, waiting to be processed by the Logstash indexer instance.

**Comment by suyograo** [comment]
LGTM, please merge

**Pull request closed without merging**